packages/forms/src/Components/Concerns/CanBeLengthConstrained.php

```diff
@@ -4,12 +4,33 @@
 
 use Closure;
 
+
 trait CanBeLengthConstrained
 {
+    protected int | Closure | null $length = null;
+
     protected int | Closure | null $maxLength = null;
 
     protected int | Closure | null $minLength = null;
 
+    public function length(int | Closure $length): static
+    {
+        $this->length = $length;
+        $this->maxLength = $length;
+        $this->minLength = $length;
+
+        $this->rule(function (): string {
+            $length = $this->getLength();
+
+            if (method_exists($this, 'isNumeric') && $this->isNumeric())
```

Comment:
Maybe we can refactor this to a `Components\Contracts\CanHaveNumericState` interface for the text input? I like to avoid `method_exists()` checks where possible.